File: solved/102/102.py
import sys

lines = sys.stdin.readlines()
for line in lines:
	arr = [int(x) for x in line.split()]
	colors = ['B', 'G', 'C']
	bins = [dict((colors[i % 3], arr[i]) for i in range((x-3), x)) for x in range(3, 12, 3)]
	final = {}
	for possibility in ['BGC', 'BCG', 'GBC', 'GCB', 'CGB', 'CBG']:
		final[possibility] = sum([sum([bins[i][x] for x in possibility if x != possibility[i]]) for i in range(3)])
	value = min(final.values())
	keys = [pair[0] for pair in final.items() if pair[1] == value]
	keys.sort()
	key = keys[0]
	print "%s %d" % (key, value)

File: solved/102/102.cc
#include <iostream>
#include <map>

using namespace std;

int main() {
	int a;
	char colors[] = {'B', 'G', 'C'};
	const char *possibilities[] = {"BGC", "BCG", "GBC", "GCB", "CGB", "CBG"};
	while(true) {
		map<const char, unsigned int> bins[3];
		for(int i = 0; i < 3; i++) {
			for(int j = 0; j < 3; j++) {
				if(!(cin >> a))
					return 0;
				bins[i][colors[j]] = a;
			}
		}
		unsigned int minSum = -1;
		const char *str;
		for(int i = 0; i < 6; i++) {
			const char *possibility = possibilities[i];
			unsigned int sum = 0;
			for(int j = 0; j < 3; j++) {
				for(int k = 0; k < 3; k++) {
					char c = possibility[k];
					if(c != possibility[j]) {
						sum += bins[j][c];
					}
				}
			}
			if((minSum == -1) || (sum < minSum) || ((sum == minSum) && string(possibility) < string(str))) {
				minSum = sum;
				str = possibility;
			}
		}
		cout << str << " " << minSum << endl;
	}
	return 0;
}